### DISCUSSION TITLE
No module names 'Hit' and 'can't find z library'

### DISCUSSION THREAD

**Discussion by AnthonyB08** [neams-th-coe/cardinal#575] [Q&A] [answered]
Hi all, 

I cloned the cardinal repository and began running the commands : 

./contrib/moose/scripts/update_and_rebuild_petsc.sh
./contrib/moose/scripts/update_and_rebuild_libmesh.sh

when I run the first commander, all seems well until the output ask me to check the libraries as seen below: 
![image](https://user-images.githubusercontent.com/55299169/217647272-425e6599-ac8a-47af-a3b3-0c15be45d5a1.png)

I run the code "make PETSC_DIR=/home/anthonyb08/cardinal/contrib/moose/scripts/../petsc PETSC_ARCH=arch-moose check"and I get a bunch of outputs saying "/usr/bin/bash: /home/anthonyb08/cardinal/contrib/moose/libmesh/installed//contrib/bin/libmesh-config: No such file or directory" 

IF I ignore this error and proceed with the next line of code : "./contrib/moose/scripts/update_and_rebuild_libmesh.sh" the output returns another error being 
![image](https://user-images.githubusercontent.com/55299169/217648942-64deb2de-6f85-4b9b-b2cd-f851d7f8e77e.png) 

What does the community recommend? ( I am working to debug this as well)

**Comment by aprilnovak** [comment, marked as answer]
Yeah, the message reported by PETSc is a bit misleading. They want you to first cd into the `petsc` directory, and _then_ do that command. 

[We have instructions here](https://cardinal.cels.anl.gov/without_conda.html#:~:text=if%20you%20want%20to%20check%20the%20petsc%20install%2C%20you%20can%20run%20the%20petsc%20tests.)

**Reply by aprilnovak** [reply]
@milljm or @loganharbour - do you think it'd be possible to add more info to that PETSc test print? Many Cardinal users have hit this before (not knowing that you need to run the suggested command from a specific location).

**Comment by aprilnovak** [comment]
Because we're getting errors during the PETSc/libMesh stage, this tells me that the error is probably not specific to Cardinal. Regarding your libMesh error, would you please re-post it on the[ MOOSE discussions page](https://github.com/idaholab/moose/discussions)? I don't know the solution to your specific error, but the MOOSE team should be able to help.

**Reply by AnthonyB08** [reply]
I believe i fixed the secondary issue by the following commander : sudo apt-get install zlib1g-dev libhdf5-dev libnetcdf-dev
I will update once compiler ends ( it has gone farther after running this install)
